java/core/src/main/java/sleeper/core/properties/table/TableProperty.java

```diff
@@ -288,6 +289,12 @@ static List<TableProperty> getAllInGroup(PropertyGroup group) {
                     "instance property.")
             .propertyGroup(TablePropertyGroup.DATA_STORAGE)
             .build();
+    TableProperty RETAIN_DATA_AFTER_DELETE = Index.propertyBuilder("sleeper.table.retain.data.after.delete")
+            .defaultProperty(RETAIN_DATA_AFTER_TABLE_REMOVAL)
+            .description("If true, removing the table from the properties file will take the table offline.\n" +
+                    "If false, it will delete all data associated with the table when the table is removed from the properties.")
```

Comment:
I think we need to clarify this is when we have an instance configuration, and we remove the table from that configuration. This applies with a set of local configuration files, or when passing a configuration directly in the CDK.

I think we need to explain the danger of doing this accidentally. It could be that you deleted the table, but it could just be that you changed the table name, and by default the system will see that as a delete and a new table.

We can make the instance property description match this more directly.